src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -0,0 +1,35 @@
+import type { QueryRawOptions, Sequelize } from '../../sequelize';
+import type { AbstractQueryGenerator, CreateSchemaQueryOptions } from './query-generator';
+
+export interface QueryInterfaceOptions {
+  sequelize: Sequelize;
+  queryGenerator: AbstractQueryGenerator;
+}
+
+export interface CreateSchemaOptions extends CreateSchemaQueryOptions, QueryRawOptions {}
+
+export class AbstractQueryInterfaceTypeScript {
+  readonly sequelize: Sequelize;
+  readonly queryGenerator: AbstractQueryGenerator;
+
+  constructor(options: QueryInterfaceOptions) {
+    this.sequelize = options.sequelize;
+    this.queryGenerator = options.queryGenerator;
+  }
+
+  /**
+   * Create a new database schema.
```

Comment:
So far our jsdoc has described methods in the third person

```suggestion
   * Creates a new database schema.
```